js/compiler_tests.ts

```diff
@@ -0,0 +1,153 @@
+  moduleName: string | null;
+  filename: string | null;
+  sourceCode: string | null;
+  outputCode: string | null;
+}
+
+function mockModuleInfo(
+  moduleName: string | null,
+  filename: string | null,
+  sourceCode: string | null,
+  outputCode: string | null
+): ModuleInfo {
+  return {
+    moduleName,
+    filename,
+    sourceCode,
+    outputCode
+  };
+}
+
+const moduleMap: {
+  [containFile: string]: { [moduleSpecifier: string]: ModuleInfo };
+} = {
+  "/root/project": {
+    "foo/bar.ts": mockModuleInfo(
+      "foo/bar",
+      "/root/project/foo/bar.ts",
+      "console.log('bar');",
+      null
+    )
+  }
+};
+
+const emittedFiles = {
+  "/root/project/foo/bar.ts": "console.log('foo');"
+};
+
+// tslint:disable:no-any
+function globalEvalMock(x: string): any {
+  console.log("globalEval", x);
+}
+function logMock(...args: any[]): void {
+  console.log(...args);
+}
+const osMock: Os = {
+  codeCache(filename: string, sourceCode: string, outputCode: string): void {
+    console.log("codeCache", { filename, sourceCode, outputCode });
+  },
+  codeFetch(moduleSpecifier: string, containingFile: string): ModuleInfo {
+    console.log("codeFetch", { moduleSpecifier, containingFile });
+    if (containingFile in moduleMap) {
+      if (moduleSpecifier in moduleMap[containingFile]) {
+        return moduleMap[containingFile][moduleSpecifier];
+      }
+    }
+    return mockModuleInfo(null, null, null, null);
+  },
+  exit(code: number): never {
+    throw new Error(`os.exit(${code})`);
+  }
+};
+// tslint:enable:no-any
+
+const tsMock: Ts = {
+  createLanguageService(host: ts.LanguageServiceHost): ts.LanguageService {
+    return {} as ts.LanguageService;
+  },
+  formatDiagnosticsWithColorAndContext(
+    diagnostics: ReadonlyArray<ts.Diagnostic>,
+    host: ts.FormatDiagnosticsHost
+  ): string {
+    return "";
+  }
+};
+const serviceMock = {
+  getCompilerOptionsDiagnostics(): ts.Diagnostic[] {
+    return [];
+  },
+  getEmitOutput(fileName: string): ts.EmitOutput {
+    console.log("getEmitOutput", fileName);
+    if (fileName in emittedFiles) {
+      return {
+        // tslint:disable-next-line:no-any
+        outputFiles: [{ text: emittedFiles[fileName] }] as any,
+        emitSkipped: false
+      };
+    }
+    return { outputFiles: [], emitSkipped: false };
+  },
+  getSemanticDiagnostics(fileName: string): ts.Diagnostic[] {
+    return [];
+  },
+  getSyntacticDiagnostics(fileName: string): ts.Diagnostic[] {
+    return [];
+  }
+};
+const windowMock = {
+  define: undefined
+};
+const mocks = {
+  _globalEval: globalEvalMock,
+  _log: logMock,
+  _os: osMock,
+  _ts: tsMock,
+  _service: serviceMock,
+  _window: windowMock
+};
+
+// tslint:disable:no-any
+const originals: { [prop: string]: any } = {};
+// tslint:enable:no-any
+
+let compiler: DenoCompiler;
+
+test(function compiler_exists() {
+  assert(DenoCompiler != null);
+  assert(DenoCompiler.instance() != null);
+});
+
+test(function compiler_tests_setup() {
```

Comment:
Please continue to use camelCase for tests. Prefix all compiler test names with `compiler`.

(I think unit_tests isn't doing this properly)

Reply:
It isn't... it was 🐵👁🐵do on my part. 🤣